util/scale_run.sh

```diff
@@ -0,0 +1,335 @@
+##########
+
+WORK_DIR="$HOME/gatling"
+PREFIX=$(date -u +%Y%m%d%H%M%S)
+
+
+##########
+# Function Definitions
+##########
+
+# Display usage help
+usage () {
+    echo "Usage: $0 [OPTION]... PE_VERSION"
+    echo
+    echo "Run scale testing on specified PE_VERSION."
+    echo "  PE_VERSION in semver notation (e.g 2019.1.0)"
+    echo
+    echo "Mandatory arguments to long options are mandatory for short options too."
+    echo "  -h|--help             show this help text"
+    echo "  -n|--noop             noop mode: prints ENV values and commands to be executed, but does not run anything"
+    echo "  -i|--run-id           Set a value to prepend to scale run directories"
+    echo "                        (default: '')"
+    echo "  --[no]tune            apply pe-tune to deployment"
+    echo "                        (default: 'true')"
+    echo
+}
+
+# Noop function to validate that bash subshells inherit
+# expected environment variables.
+function echo_env () {
+    echo "============================================
+Provided args:
+TUNE=$TUNE
+RUN_ID=$RUN_ID
+PE_VERSION=$PE_VERSION
+
+In subshell for $test $run_type round $i
+cmd=$cmd
+BEAKER_INSTALL_TYPE=$BEAKER_INSTALL_TYPE
+BEAKER_PE_DIR=$BEAKER_PE_DIR
+BEAKER_PE_VER=$BEAKER_PE_VER
+PUPPET_GATLING_SCALE_BASE_INSTANCES=$PUPPET_GATLING_SCALE_BASE_INSTANCES
+PUPPET_GATLING_SCALE_ITERATIONS=$PUPPET_GATLING_SCALE_ITERATIONS
+PUPPET_GATLING_SCALE_INCREMENT=$PUPPET_GATLING_SCALE_INCREMENT
+PUPPET_GATLING_SCALE_SCENARIO=$PUPPET_GATLING_SCALE_SCENARIO
+ABS_AWS_MOM_SIZE=$ABS_AWS_MOM_SIZE
+============================================"
+}
+
+# Prepare checkout of puppet-gatling-load-test
+function prep_gplt () {
+    mkdir -p "$WORK_DIR"
+    cd "$WORK_DIR" || exit 1
+    git clone git@github.com:puppetlabs/gatling-puppet-load-test "$1"
+    cd "$1" || exit 1
+    bundle install --path vendor/bundle
+}
+
+
+##########
+# Argument checking
+##########
+
+ARGS=()
+while [ $# -gt 0 ]
+do
+    case "$1" in
+        -h|--help)  usage
+            exit 1
+            ;;
+        -n|--noop)
+            NOOP=true
+            shift
+            ;;
+        -i|--run-id)
+          RUN_ID=$2
+          PREFIX="$PREFIX-$RUN_ID"
+          shift 2
+          ;;
+        --tune)
+          TUNE=true
+          shift
+          ;;
+        --notune)
+          TUNE=false
+          shift
+          ;;
+        --) # end argument parsing
+            shift
+            break
+            ;;
+        -*|--*=) # unsupported flags
+            echo "Error: Unsupported flag $1" >&2
+            exit 1
+            ;;
+        *) # preserve positional arguments
+            ARGS+=("$1")
+            shift
+            ;;
+        esac
+done
+
+# Ensure that TUNE is set to true by default
+TUNE="${TUNE:-true}"
+
+# Ensure PE_VERSION is provided
+if [ "${#ARGS[@]}" -le "0" ]; then
+    echo "PE_VERSION not provided"
+    echo
+    usage
+    exit 1
+fi
+PE_VERSION=${ARGS[0]}
+
+# Ensure PE_VERSION is semver
```

Comment:
This doesn't seem to work when running with the -n option:
```
bill.claytor:~/RubymineProjects/_pr/SLV-414/gatling-puppet-load-test> ./util/scale_run.sh -n -i SLV-414 testing123
PE_MAJOR=testing123
PE_MINOR=testing123
PE_PATCH=testing123
PE_BUILD=testing123
```

Reply:
I have patched this bug.

Here are some tests
```
$ ./util/scale_run.sh -n 2019.1.1-rc0-207-g4e47830
PE_MAJOR=2019
PE_MINOR=1
PE_PATCH=1
PE_BUILD=-rc0-207-g4e47830
...
$ ./util/scale_run.sh -n 1.1.1
PE_MAJOR=1
PE_MINOR=1
PE_PATCH=1
PE_BUILD=
...
$ ./util/scale_run.sh -n 1.1
PE_MAJOR=
PE_MINOR=
PE_PATCH=
PE_BUILD=
Supplied PE_VERSION (1.1) is malformed
...
$ ./util/scale_run.sh -n 1.1.
PE_MAJOR=1
PE_MINOR=1
PE_PATCH=
PE_BUILD=
Supplied PE_VERSION (1.1.) is malformed
...
 $ ./util/scale_run.sh -n testing123
PE_MAJOR=
PE_MINOR=
PE_PATCH=
PE_BUILD=
Supplied PE_VERSION (testing123) is malformed
```